worker-policy/src/main/java/com/github/cafdataprocessing/worker/policy/ExecuteTaskData.java

```diff
@@ -670,15 +672,17 @@ private boolean canSkipAlreadyCompletedSequence(com.github.cafdataprocessing.cor
      */
     private void registerHandlers(String projectId) {
         if (classifyDocumentApi instanceof ClassifyDocumentApiDirectImpl) {
-            UserContext userContext = corePolicyApplicationContext.getBean(UserContext.class);
+            final UserContext userContext = corePolicyApplicationContext.getBean(UserContext.class);
             userContext.setProjectId(projectId);
-            if (!registeredUsers.contains(projectId)) {
-                ServiceLoader<WorkerPolicyHandler> loader = ServiceLoader.load(WorkerPolicyHandler.class);
+            synchronized (REGISTERED_USERS_LOCK) {
+                if (!registeredUsers.contains(projectId)) {
```

Comment:
Should we not also be checking this condition before the synchronize so that we avoid every task thread waiting and then running that if condition to see it is false and they don't have to setup handlers for the tenant?